drivers/spi/spi_sc18is606.c

```diff
@@ -0,0 +1,278 @@
+		return -ENOTSUP;
+	}
+
+	const int bits = SPI_WORD_SIZE_GET(config->operation);
+
+	if (bits > 8) {
+		LOG_ERR("Word sizes > 8 bits not supported");
+		return -ENOTSUP;
+	}
+
+	/* Build SC18IS606  configuration byte*/
+	uint8_t cfg_byte = 0;
+
+	cfg_byte |= ((config->operation & SPI_TRANSFER_LSB) >> 4) << 5;
+
+	uint8_t mode = (SPI_MODE_GET(config->operation) >> 1) & 0x3;
+
+	cfg_byte |= (mode << 2);
+
+	cfg_byte |= (config->frequency & 0x3);
+
+	data->ctx.config = config;
+
+	uint8_t buffer[2] = {SC18IS606_CONFIG_SPI, cfg_byte};
+
+	cfg_byte = nxp_sc18is606_transfer(dev, buffer, sizeof(buffer), NULL, 0, NULL);
+
+	return 0;
+}
+
+static int sc18is606_spi_transceive(const struct device *dev, const struct spi_config *spi_cfg,
+				    const struct spi_buf_set *tx_buffer_set,
+				    const struct spi_buf_set *rx_buffer_set)
+{
+	int ret = 1;
+
+	ret = sc18is606_spi_configure(dev, spi_cfg);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (!tx_buffer_set && !rx_buffer_set) {
+		LOG_ERR("SC18IS606 at least one buffer_set should be set");
+		return -EINVAL;
+	}
+
+	/* CS line to be Used */
+	uint8_t ss_idx = spi_cfg->slave;
+
+	if (ss_idx > 2) {
+		LOG_ERR("SC18IS606: Invalid SS Index (%u) must be 0-2", ss_idx);
+		return -EINVAL;
+	}
+
+	uint8_t function_id = (1 << ss_idx) & 0x07;
+
+	if (tx_buffer_set && tx_buffer_set->buffers && tx_buffer_set->count > 0) {
+		for (size_t i = 0; i < tx_buffer_set->count; i++) {
+			const struct spi_buf *tx_buf = &tx_buffer_set->buffers[i];
+
+			uint8_t id_buf[1] = {function_id};
+
+			ret = nxp_sc18is606_transfer(dev, tx_buf->buf, sizeof(tx_buf), NULL, 0,
+						     id_buf);
+			if (ret < 0) {
+				LOG_ERR("SC18IS606: TX of size: %d failed %s", tx_buf->len,
+					dev->name);
+				return ret;
+			}
+		}
+	}
+
+	if (rx_buffer_set && rx_buffer_set->buffers && rx_buffer_set->count > 0) {
+		for (size_t i = 0; i < rx_buffer_set->count; i++) {
+			/* Function ID first to select the device */
+			uint8_t cmd_buf[1] = {function_id};
+
+			const struct spi_buf *rx_buf = &rx_buffer_set->buffers[i];
+
+			ret = nxp_sc18is606_transfer(dev, cmd_buf, sizeof(cmd_buf), rx_buf->buf,
+						     rx_buf->len, NULL);
+
+			if (ret < 0) {
+				LOG_ERR("SC18IS606: RX of size: %d failed on  (%s)", rx_buf->len,
+					dev->name);
+				return ret;
+			}
+		}
+	}
+
+	return ret;
+}
+
+int sc18is606_spi_release(const struct device *dev, const struct spi_config *config)
+{
+	struct nxp_sc18is606_data *data = dev->data;
+
+	struct spi_context *ctx = &data->ctx;
+
+	spi_context_unlock_unconditionally(ctx);
+
+	return 0;
+}
+
+static DEVICE_API(spi, sc18is606_api) = {
+	.transceive = sc18is606_spi_transceive,
+	.release = sc18is606_spi_release,
+};
+
+static int sc18is606_init(const struct device *dev)
+{
+	const struct nxp_sc18is606_config *cfg = dev->config;
+	struct nxp_sc18is606_data *data = dev->data;
+	int ret;
+
+	struct spi_config my_config = {
+		.frequency = data->frequency,
+		.operation = data->spi_mode,
+		.slave = data->spi_mode,
```

Comment:
typo? should be cs / slave index?

Reply:
@kartben  I am using slave here because that's what I am getting from the device tree, so I can control the ss lines on this bridge